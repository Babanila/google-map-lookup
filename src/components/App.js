import React from 'react'
import { cx, css } from 'emotion'

const rootDiv = css`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
`

function App() {
  return <div className={cx(rootDiv)}>Testing Setup</div>
}

export default App
