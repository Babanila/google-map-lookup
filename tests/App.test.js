import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import App from '../src/components/App'

describe('Check App Component', () => {
  it('should renders App without crashing', () => {
    const tree = shallow(<App />)
    expect(tree).toMatchSnapshot()
  })

  it('should render App component with data', () => {
    const AppWrapper = mount(
      <MemoryRouter initialentries="{['/']}">
        <App />
      </MemoryRouter>
    )

    const actualMemoryRouter = AppWrapper.find('MemoryRouter')
    const actualRouter = actualMemoryRouter.find('Router')
    const actualComponent = actualRouter.find('div')

    expect(AppWrapper).toBeDefined()
    expect(actualMemoryRouter.length).toEqual(1)
    expect(actualRouter.length).toEqual(1)
    expect(actualComponent.length).toEqual(3)
  })
})
