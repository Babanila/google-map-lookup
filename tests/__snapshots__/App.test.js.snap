// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check App Component should renders App without crashing 1`] = `
<Component
  className="css-1edrwtt"
>
  <PageHeader />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/location/:postcode"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Component>
`;
